File: Workout.cs
using System;

namespace HealthTrackerApp
{
    public class Workout
    {
        public DateTime Date { get; set; }
        public string ExerciseType { get; set; }
        public int Duration { get; set; }  // Duration in minutes
        public int Calories { get; set; }
        public string Notes { get; set; }

        public Workout()
        {
        }

        public override string ToString()
        {
            int hours = Duration / 60;
            int minutes = Duration % 60;
            string duration = hours > 0 
                ? $"{hours}h {minutes}m" 
                : $"{minutes}m";
            return $"{Date.ToShortDateString()} - {ExerciseType} ({duration}, {Calories} cal)";
        }
    }
} 

File: HealthTrackerQt/workout.cpp
#include "workout.h"

Workout::Workout(const QDateTime& date, const QString& exerciseType, int durationMinutes, int caloriesBurned, const QString& notes)
    : m_date(date)
    , m_exerciseType(exerciseType)
    , m_durationMinutes(durationMinutes)
    , m_caloriesBurned(caloriesBurned)
    , m_notes(notes)
{
}

QString Workout::toString() const
{
    return QString("%1 - %2 (%3 min, %4 cal)")
        .arg(m_date.toString("MM/dd/yyyy"))
        .arg(m_exerciseType)
        .arg(m_durationMinutes)
        .arg(m_caloriesBurned);
} 